### PR TITLE
PHP 8.0 support #20

### LINKED ISSUE #20: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by svycka**
# Feature Request

Q | A
-- | --
New Feature | yes

Summary
To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [x] Modify composer.json to provide support for PHP 8.0 by adding the constraint ~8.0.0
- [x] Modify composer.json to drop support for PHP less than 7.3
- [x] Modify composer.json to implement phpunit 9.3 which supports PHP 7.3+
- [x] Modify .travis.yml to ignore platform requirements when installing composer dependencies (simply add --ignore-platform-reqs to COMPOSER_ARGS env variable)
- [x] Modify .travis.yml to add PHP 8.0 to the matrix (NOTE: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [x] Modify source code in case there are incompatibilities with PHP 8.0

closes #20

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,16 +25,17 @@
         }
     },
     "require": {
-        "php": "^5.6 || ^7.0",
+        "php": "^7.3 || ~8.0.0",
         "laminas/laminas-filter": "^2.9.1",
-        "laminas/laminas-servicemanager": "^2.7.10 || ^3.3.1",
```
Comment (svycka) [inline composer.json:30]: because SM v2 does not support php8 and PHPUnit ^9

Comment (michalbundyra) [reply composer.json:30]: It's fine to drop SM v2, as it is no longer supported.

**Comment by svycka** [comment]
@boesing then I think it's ready for review

File: test/InputFilterAwareTraitTest.php [hunk 2]
```diff
@@ -22,13 +22,16 @@ public function testSetInputFilter()
     {
         $object = $this->getObjectForTrait(InputFilterAwareTrait::class);
 
-        $this->assertAttributeEquals(null, 'inputFilter', $object);
+        $r = new \ReflectionObject($object);
```
Comment (samsonasik) [inline test/InputFilterAwareTraitTest.php]: can be moved to use statement

**Review by boesing — APPROVED** [review]
LGTM

**Comment by FabianKoestring** [comment]
Could you tag a new release for this feature?

**Comment by samsonasik** [comment]
released 2.11.0 https://github.com/laminas/laminas-inputfilter/releases/tag/2.11.0

**Pull request merged**